canary/producer-cpp/src/Include.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <gst/gst.h>
+#include <gst/app/gstappsink.h>
+#include <string.h>
+#include <chrono>
+#include <Logger.h>
+#include "KinesisVideoProducer.h"
+#include <vector>
+#include <stdlib.h>
+#include <mutex>
+#include <IotCertCredentialProvider.h>
+#include <unistd.h> 
+#include <com/amazonaws/kinesis/video/cproducer/Include.h>
+#include <aws/core/Aws.h>
+#include <aws/monitoring/CloudWatchClient.h>
+#include <aws/monitoring/model/PutMetricDataRequest.h>
+#include <aws/logs/CloudWatchLogsClient.h>
+#include <aws/logs/model/CreateLogGroupRequest.h>
+#include <aws/logs/model/CreateLogStreamRequest.h>
+#include <aws/logs/model/PutLogEventsRequest.h>
+#include <aws/logs/model/DeleteLogStreamRequest.h>
+#include <aws/logs/model/DescribeLogStreamsRequest.h>
+
+#include "CanaryCallbackProvider.h"
+#include "CanaryConfig.h"
+#include "CanaryLogs.h"
+#include "CustomData.h"
+
+
+using namespace std;
+using namespace com::amazonaws::kinesis::video;
+using namespace log4cplus;
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int gstreamer_init(int, char **);
+
+#ifdef __cplusplus
+}
+#endif
+
+LOGGER_TAG("com.amazonaws.kinesis.video.gstreamer");
+
+#define DEFAULT_RETENTION_PERIOD_HOURS 2
+#define DEFAULT_KMS_KEY_ID ""
+#define DEFAULT_STREAMING_TYPE STREAMING_TYPE_REALTIME
+#define DEFAULT_CONTENT_TYPE "video/h264"
+#define DEFAULT_MAX_LATENCY_SECONDS 60
+// #define DEFAULT_FRAGMENT_DURATION_MILLISECONDS 2000
+#define DEFAULT_TIMECODE_SCALE_MILLISECONDS 1
+#define DEFAULT_KEY_FRAME_FRAGMENTATION TRUE
+#define DEFAULT_FRAME_TIMECODES TRUE
+#define DEFAULT_ABSOLUTE_FRAGMENT_TIMES TRUE
+#define DEFAULT_FRAGMENT_ACKS TRUE
+#define DEFAULT_RESTART_ON_ERROR TRUE
+#define DEFAULT_RECALCULATE_METRICS TRUE
+#define DEFAULT_AVG_BANDWIDTH_BPS (4 * 1024 * 1024)
+// #define DEFAULT_BUFFER_DURATION_SECONDS 120
```

Comment:
Remove if not required. Also, If we plan to have an Include.h, I would suggest moving #defines from all the files into this and have all header files reference Include.h file. It is more modular and cleaner.

Reply:
Having Include.h file in all the other headers would not work because they are all being aggregated to be included into the Include.h file - this aggregation is necessary because there are classes that refer to other classes.